components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -559,7 +588,27 @@ interface ReadableHistoryConnection : InterruptibleConnection {
      * @param start The (inclusive) start time to bound the query.
      * @param end The (inclusive) end time to bound the query.
      */
-    fun getVisitInfos(start: Long, end: Long = Long.MAX_VALUE): List<VisitInfo>
+    fun getVisitInfos(
+        start: Long,
+        end: Long = Long.MAX_VALUE,
+        excludeTypes: List<VisitType> = listOf()
+    ): List<VisitInfo>
+
+    /**
+     * Return a "page" of history results.
+     *
+     * @param offset The offset where the page begins.
+     * @param count The number of items to return in the page.
+     * @param excludeTypes List of visit types to exclude.
+     */
+    fun getVisitPage(offset: Long, count: Long, excludeTypes: List<VisitType> = listOf()): List<VisitInfo>
+
+    /**
+     * Get the number of history visits.
+     *
+     * @param excludeTypes List of visit types to exclude.
+     */
+    fun getVisitCount(excludeTypes: List<VisitType> = listOf()): Long
```

Comment:
It's interesting that in between `getVisitCount` and `getVisitPage` sync might have occurred, and the two will be out-of-sync without a way to know that something's off. Not sure if we would want to deal with this in practice. `getVisitPage` could take `expectedCount` value, for example, to enforce things. But that will make this API more convoluted to use and will likely have a perf penalty.

So perhaps just make a note of this in the docs?